### PR TITLE
Raise error when passed invalid config file path

### LINKED ISSUE #362: Providing invalid config file path should raise instead of silently ignoring it
RIght now, it's possible to provide an invalid config file path and ameba just ignores it and uses the default config.
Instead it should raise, and abort the execution altogether.

### PR DESCRIPTION
**Pull request by stufro**
Resolves #362 

I'm new here, let me know if you think this could be done in a different place/better way.

### CODE REVIEW

File: spec/ameba/cli/cmd_spec.cr [hunk 1]
```diff
@@ -20,8 +20,14 @@ module Ameba::Cli
 
       %w(-c --config).each do |f|
         it "accepts #{f} flag" do
-          c = Cli.parse_args [f, "config.yml"]
-          c.config.should eq Path["config.yml"]
+          c = Cli.parse_args [f, "shard.yml"]
+          c.config.should eq Path["shard.yml"]
```
Comment (stufro) [inline spec/ameba/cli/cmd_spec.cr:24]: I had to change the config file we pass to `parse_args` to a file which actually exists so I chose the shard file.

I could create a temp file in the test and throw it away afterwards?

File: src/ameba/cli/cmd.cr [hunk 2]
```diff
@@ -77,6 +77,8 @@ module Ameba::Cli
 
       parser.on("-c", "--config PATH",
         "Specify a configuration file") do |path|
+        raise ArgumentError.new("Unable to find config file #{path}") if !File.exists?(path) && !opts.skip_reading_config?
+
```
Comment (veelenga) [inline src/ameba/cli/cmd.cr:81]: Hey and thanks for your contribution. 

can we move this logic to the config itself? 

https://github.com/crystal-ameba/ameba/blob/8c9d234d0b06d945d0cf841dfa0596734acc715c/src/ameba/config.cr#L109-L128

File: spec/ameba/config_spec.cr [hunk 3]
```diff
@@ -84,6 +84,12 @@ module Ameba
         config.formatter.should_not be_nil
       end
 
+      it "raises when custom config file doesn't exist" do
+        expect_raises(Exception, "Config file is invalid: Unable to find config file foo.yml") do
+          Config.load "foo.yml"
+        end
```
Comment (veelenga) [inline spec/ameba/config_spec.cr:90]: can we rephrase the error message re-thrown from `load` method now? Maybe something like

`Config file is invalid` -> `Unable to load config file`

Previously the only reason why it was thrown is invalid YAML. Now the file may not exist here.

File: src/ameba/config.cr [hunk 4]
```diff
@@ -120,7 +120,7 @@ class Ameba::Config
 
   protected def self.read_config(path = nil)
     if path
-      return File.exists?(path) ? File.read(path) : nil
+      return File.exists?(path) ? File.read(path) : raise("Unable to find config file #{path}")
```
Comment (veelenga) [inline src/ameba/config.cr:123]:
```suggestion
      raise ArgumentError.new("Config file does not exist #{path}") unless File.exists?(path)
      return File.read(path)
```

File: spec/ameba/config_spec.cr (hunk 3 above)
Comment (stufro) [reply spec/ameba/config_spec.cr]: sure thing good shout :) 4741c9f

File: spec/ameba_fixture.yml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+Lint/ComparisonToBoolean:
+  Enabled: true
```
Comment (veelenga) [inline spec/ameba_fixture.yml:2]: could you please move it spec/fixtures/config.yml ?

**Review by veelenga — COMMENTED** [review]
The last minor comment. The rest of the changes LGTM

**Review by veelenga — APPROVED** [review]
LGTM

**Comment by Sija** [comment]
This change forces every glob path under the `Globs` and `Excluded` keys to match, which is a breaking change -IMO unwanted. We need to either address that or revert this PR altogether.

**Pull request merged**